spec/unit/provider/service/windows_spec.rb

```diff
@@ -417,37 +426,19 @@
     include_context "testing private methods"
 
     let(:username) { "unit_test_user" }
-    let(:success_string) { "The task has completed successfully.\r\nSee logfile etc." }
-    let(:failure_string) { "Look on my works, ye Mighty, and despair!" }
-    let(:command) {
-      dbfile = provider.grant_dbfile_name(username)
-      policyfile = provider.grant_policyfile_name(username)
-      logfile = provider.grant_logfile_name(username)
-
-      %Q{secedit.exe /configure /db "#{dbfile}" /cfg "#{policyfile}" /areas USER_RIGHTS SECURITYPOLICY SERVICES /log "#{logfile}"}
-    }
-    let(:shellout_env) { {:environment=>{"LC_ALL"=>"en_US.UTF-8"}} }
 
-    before {
-      expect_any_instance_of(described_class).to receive(:shell_out).with(command).and_call_original
-      expect_any_instance_of(Mixlib::ShellOut).to receive(:run_command).and_return(nil)
-    }
-
-    after {
-      # only needed for the second test.
-      ::File.delete(provider.grant_policyfile_name(username)) rescue nil
-      ::File.delete(provider.grant_logfile_name(username)) rescue nil
-      ::File.delete(provider.grant_dbfile_name(username)) rescue nil
-    }
-
-    it "calls Mixlib::Shellout with the correct command string" do
-      expect_any_instance_of(Mixlib::ShellOut).to receive(:exitstatus).and_return(0)
+    it "calls win32 api to grant user SeServiceLogonRight" do
+      expect(Chef::ReservedNames::Win32::Security).to receive(:add_account_right).with(username, service_right)
       expect(provider.grant_service_logon(username)).to equal true
     end
 
-    it "raises an exception when the grant command fails" do
-      expect_any_instance_of(Mixlib::ShellOut).to receive(:exitstatus).and_return(1)
-      expect_any_instance_of(Mixlib::ShellOut).to receive(:stdout).and_return(failure_string)
+    it "strips '.\' from user name when sending to win32 api" do
+      expect(Chef::ReservedNames::Win32::Security).to receive(:add_account_right).with(username, service_right)
+      expect(provider.grant_service_logon(".\\#{username}")).to equal true
```

Comment:
The win32 APIs used don't understand `.\username`. However they work fine with all of these:
* `username`
* `domain\username`
* `computer\\username`

We still allow users to use the `.\username` format and strip off the `.\` here. That makes sense since its a generally accepted format elsewhere.